src/main/logutils.ts

```diff
@@ -665,46 +669,49 @@ const forceStopRecording = () => {
 }
 
 /**
- * checkWoWProcess
- * @returns {[boolean, boolean]} retailRunning, classicRunning
+ * Check Windows task list and find any WoW process.
  */
-const checkWoWProcess = async (): Promise<[boolean, boolean]> => {
-    let retailRunning = false;
-    let classicRunning = false;
-
+const checkWoWProcess = async (): Promise<IWoWProcessResult[]> => {
+    const wowProcessRx = new RegExp(/(wow(T|B|classic)?)\.exe/, 'i');
     const taskList = await tasklist();
 
-    taskList.forEach((process: any) => {
-        if (process.imageName === "Wow.exe") {
-            retailRunning = true;
-        } else if (process.imageName === "WowClassic.exe") {
-            classicRunning = true;
-        }
-    });
-
-    return [retailRunning, classicRunning]
+    return taskList
+        // Map all processes found to check if they match `wowProcessRx`
+        .map((p: any) => p.imageName.match(wowProcessRx))
+        // Remove those that result in `null` (didn't match)
+        .filter((p: any) => p)
+        // Return an object suitable for `IWoWProcessResult`
+        .map((match: any): IWoWProcessResult => ({
+            exe: match[0],
+            flavour: wowExecutableFlavours[match[1].toLowerCase()]
+        }))
+    ;
 }
 
 /**
  * pollWoWProcessLogic
  */
-const pollWoWProcessLogic = async (startup: boolean) => {
-    const [retailFound, classicFound] = await checkWoWProcess();
-    const retailProcessChanged = (retailFound !== isRetailRunning);
-    // TODO classic support
-    const classicProcessChanged = (classicFound !== isClassicRunning);
-    const processChanged = (retailProcessChanged || classicProcessChanged);
-    if (!retailProcessChanged && !startup) return;
-    (retailFound) ? wowProcessStarted() : wowProcessStopped();
+const pollWoWProcessLogic = async () => {
+    const wowProcesses = await checkWoWProcess();
+    const wowProcess = wowProcesses.pop();
+
+    if (wowProcess && wowProcessRunning === null) {
+        wowProcessStarted(wowProcess);
+    } else
+    if (!wowProcess && wowProcessRunning !== null) {
+        wowProcessStopped();
+    }
 }
 
 /**
  * pollWoWProcess
  */
 const pollWowProcess = () => {
-    pollWoWProcessLogic(true);
```

Comment:
What was the point of this logic of `pollWoWProcessLogic()` not starting or stopping if its argument was `false`?
I've removed it because it seemed pointless and the app responds as expected without it, but if there was a specific reason for it that I don't know about, I'll revert that change.

Reply:
I'm not really sure actually. I think it can be removed. Vaguely remember being irritated by the log we made as "Wow has started" isn't really true if we've just started WR with wow already running. But as far as I can tell we still suffer for that and just the log is a little less specific with "wow is running".

Reply:
Yeah I was considering changing the log message to "_Detected WoW (<flavour>)_"

Reply:
Sounds good to me!